### PR TITLE
fix customized hub namespace name cause secret not found when do csr sign

### PR DESCRIPTION
**Pull request by xuezhaojun**
Signed-off-by: xuezhaojun <zxue@redhat.com>

### CODE REVIEW

File: pkg/hub/addon/addon.go [hunk 1]
```diff
@@ -216,7 +222,12 @@ func (a *clusterProxyAddOnAgent) csrConfigurations(cluster *clusterv1.ManagedClu
 func (a *clusterProxyAddOnAgent) sign(csr *certificatesv1.CertificateSigningRequest) []byte {
 	// We consider approved csr no need to do verify again
 	klog.Infof("CSRSign, agent: %s", config.ADDON_AGENT_NAME)
-	s, err := a.hubKubeClient.CoreV1().Secrets(config.HUB_NAMESPACE).Get(context.TODO(), config.SignerSecret, metav1.GetOptions{})
+	hubNamespace, err := utils.GetComponentNamespace()
+	if err != nil {
+		klog.Errorf("get component namespace failed, err:%v", err)
```
Comment (qiujian16) [inline pkg/hub/addon/addon.go:227]: you might just want to warn here and fall back to use the open-cluster-management ns here?

Comment (skeeey) [reply pkg/hub/addon/addon.go:227]: I think you may put the current component namespace in the `clusterProxyAddOnAgent` struct, when you new the addon agent, you pass the current namespace to it

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package utils
+
+import "io/ioutil"
+
+func GetComponentNamespace() (string, error) {
+	nsBytes, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "open-cluster-management", err
```
Comment (skeeey) [inline pkg/utils/utils.go:8]: you may don't need return err, just fall back to the default namespace

Comment (skeeey) [reply pkg/utils/utils.go:8]: and we already have this in the `helpers` https://github.com/open-cluster-management/cluster-proxy-addon/blob/main/pkg/helpers/helpers.go#L11, you can use it

Comment (xuezhaojun) [reply pkg/utils/utils.go:8]: Ok, sure.

File: pkg/hub/addon/addon.go (hunk 1 above)
Comment (xuezhaojun) [reply pkg/hub/addon/addon.go:227]: Sure. Using the `GetCurrentNamespace` in pkg helper with a default value now.

**Pull request merged**